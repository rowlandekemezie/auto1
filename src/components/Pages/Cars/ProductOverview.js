import React from "react";
import styled from "styled-components";
import PlaceHolder from "react-placeholder";
import PropTypes from "prop-types";
import Link from "react-router-dom/Link";

import FilterBox from "../../FilterBox";
import Pagination from "../../Pagination";

import { capitalize } from "../../../utils";

const Wrapper = styled("div")`
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    width: 66.666%;
  }
`;

const CardWrapper = styled("div")`
  display: flex;
  border: 2px solid ${({ theme }) => theme.GRAY};
  transition: all 0.3s ease;
  padding: 12px;
  margin-bottom: 12px;

  &:focus,
  &:hover {
    box-shadow: 5px 5px 25px 0px rgba(46, 61, 73, 0.2);
  }
`;

const Description = styled("div")`
  justify-content: center;
  light-height: 18px;
`;

const StyledLink = styled(Link)`
  color: ${({ theme }) => theme.ORANGE};
  font-size: 12px;
  text-decoration: none;
`;

const Image = styled("img")`
  width: 120px;
  height: 120px;
  margin-right: 24px;
`;

const Text = styled("p")`
  font-size: 12px;
  font-weight: 400;
`;

const Title = styled("p")`
  font-weight: 700;
  font-size: 18px;
  margin-top: 8px;
`;

const SortFilterBoxWrapper = styled("div")`
  margin-bottom: 24px;

  @media (min-width: 768px) {
    width: 40%;
  }
`;

const HeadingContainer = styled("div")`
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
  }
`;

const PageMetaData = styled("div")`
  width: 60%;
`;

const SubTitle = styled("p")`
  font-size: 18px;
`;

function Card({ car }) {
  return (
    <CardWrapper>
      <Image src={car.pictureUrl} />
      <Description>
        <Title>{car.modelName}</Title>
        <Text>
          Stock #{car.stockNumber} - {car.mileage.number} KM - {car.fuelType} -{" "}
          {capitalize(car.color)}{" "}
        </Text>
        <StyledLink to={`/cars/${car.stockNumber}`}>View details</StyledLink>
      </Description>
    </CardWrapper>
  );
}

const SORTING_OPTIONS = [
  { value: "", label: "None" },
  { value: "asc", label: "Mileage - Ascending Order" },
  { value: "des", label: "Mileage - Descending Order" }
];

export default function ProductOverview({
  cars,
  value,
  onNext,
  isLoading,
  onPrevious,
  currentPage,
  previousPage,
  totalPageCount,
  handleOnChange
}) {
  return (
    <PlaceHolder ready={!!cars || isLoading} type="media" row={7}>
      <Wrapper>
        <HeadingContainer>
          <PageMetaData>
            <Title>Available cars</Title>
            <SubTitle>
              Showing {cars.length} of {totalPageCount}{" "}
              {`result${cars.length > 1 ? "s" : ""}`}{" "}
            </SubTitle>
          </PageMetaData>
          <SortFilterBoxWrapper>
            <FilterBox
              value={value}
              options={SORTING_OPTIONS}
              title="Sort by"
              handleOnChange={handleOnChange}
            />
          </SortFilterBoxWrapper>
        </HeadingContainer>
        {cars.map((car, index) => (
          <Card car={car} key={`${car.stockNumber}-${index}`} />
        ))}
        <Pagination
          currentPage={currentPage}
          previousPage={previousPage}
          totalPageCount={totalPageCount}
          onNext={onNext}
          onPrevious={onPrevious}
        />
      </Wrapper>
    </PlaceHolder>
  );
}

ProductOverview.propTypes = {
  cars: PropTypes.arrayOf(
    PropTypes.shape({
      modelName: PropTypes.string.isRequired,
      stockNumber: PropTypes.number.isRequired,
      color: PropTypes.string.isRequired,
      fuelType: PropTypes.string.isRequired,
      mileage: PropTypes.shape({
        number: PropTypes.number.isRequired
      })
    }).isRequired
  ),
  onNext: PropTypes.func.isRequired,
  onPrevious: PropTypes.func.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  totalPageCount: PropTypes.number,
  value: PropTypes.object,
  currentPage: PropTypes.number,
  previousPage: PropTypes.number
};
