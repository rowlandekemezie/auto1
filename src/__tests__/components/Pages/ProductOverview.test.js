import React from "react";
import { shallow, mount } from "enzyme";
import { Router } from "react-router-dom";
import history from "history/createBrowserHistory";
import ProductOverview from "../../../components/Pages/Cars/ProductOverview";

describe("<ProductOverview />", () => {
  const props = {
    carId: "1234",
    fetchACar: jest.fn(),
    onNext: jest.fn(),
    onPrevious: jest.fn(),
    handleOnChange: jest.fn(),
    cars: [
      {
        modelName: "1",
        stockNumber: 2,
        color: "red",
        fuelType: "diesel",
        mileage: { number: 234 }
      }
    ]
  };

  it("should render component properly", () => {
    const wrapper = shallow(<ProductOverview {...props} />);
    expect(wrapper.length).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with the required proptypes and exact components", () => {
    const wrapper = mount(
      <Router history={history()}>
        <ProductOverview {...props} />
      </Router>
    );
    const container = wrapper.find(ProductOverview);
    expect(container.length).toBe(1);
    expect(container.find("Card").length).toBe(1);
    expect(container.find("Link").length).toBe(1);
    expect(container.props().cars).toBeArray();
    expect(container.props().cars[0].mileage).toBeObject();
  });
});
