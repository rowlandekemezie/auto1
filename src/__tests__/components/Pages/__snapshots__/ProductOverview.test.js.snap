// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductOverview /> should render component properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductOverview
    carId="1234"
    cars={
      Array [
        Object {
          "color": "red",
          "fuelType": "diesel",
          "mileage": Object {
            "number": 234,
          },
          "modelName": "1",
          "stockNumber": 2,
        },
      ]
    }
    fetchACar={[MockFunction]}
    handleOnChange={[MockFunction]}
    onNext={[MockFunction]}
    onPrevious={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef>
          <ForwardRef>
            <ForwardRef>
              Available cars
            </ForwardRef>
            <ForwardRef>
              Showing 
              1
               of 
               
              result
               
            </ForwardRef>
          </ForwardRef>
          <ForwardRef>
            <FilterBox
              handleOnChange={[MockFunction]}
              options={
                Array [
                  Object {
                    "label": "None",
                    "value": "",
                  },
                  Object {
                    "label": "Mileage - Ascending Order",
                    "value": "asc",
                  },
                  Object {
                    "label": "Mileage - Descending Order",
                    "value": "des",
                  },
                ]
              }
              title="Sort by"
            />
          </ForwardRef>
        </ForwardRef>
        <Card
          car={
            Object {
              "color": "red",
              "fuelType": "diesel",
              "mileage": Object {
                "number": 234,
              },
              "modelName": "1",
              "stockNumber": 2,
            }
          }
        />
        <Pagination
          onNext={[MockFunction]}
          onPrevious={[MockFunction]}
        />
      </ForwardRef>,
      "color": "#CDCDCD",
      "delay": 0,
      "ready": true,
      "row": 7,
      "type": "media",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <ForwardRef>
              <ForwardRef>
                Available cars
              </ForwardRef>
              <ForwardRef>
                Showing 
                1
                 of 
                 
                result
                 
              </ForwardRef>
            </ForwardRef>
            <ForwardRef>
              <FilterBox
                handleOnChange={[MockFunction]}
                options={
                  Array [
                    Object {
                      "label": "None",
                      "value": "",
                    },
                    Object {
                      "label": "Mileage - Ascending Order",
                      "value": "asc",
                    },
                    Object {
                      "label": "Mileage - Descending Order",
                      "value": "des",
                    },
                  ]
                }
                title="Sort by"
              />
            </ForwardRef>
          </ForwardRef>,
          Array [
            <Card
              car={
                Object {
                  "color": "red",
                  "fuelType": "diesel",
                  "mileage": Object {
                    "number": 234,
                  },
                  "modelName": "1",
                  "stockNumber": 2,
                }
              }
            />,
          ],
          <Pagination
            onNext={[MockFunction]}
            onPrevious={[MockFunction]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <ForwardRef>
                  Available cars
                </ForwardRef>
                <ForwardRef>
                  Showing 
                  1
                   of 
                   
                  result
                   
                </ForwardRef>
              </ForwardRef>,
              <ForwardRef>
                <FilterBox
                  handleOnChange={[MockFunction]}
                  options={
                    Array [
                      Object {
                        "label": "None",
                        "value": "",
                      },
                      Object {
                        "label": "Mileage - Ascending Order",
                        "value": "asc",
                      },
                      Object {
                        "label": "Mileage - Descending Order",
                        "value": "des",
                      },
                    ]
                  }
                  title="Sort by"
                />
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef>
                    Available cars
                  </ForwardRef>,
                  <ForwardRef>
                    Showing 
                    1
                     of 
                     
                    result
                     
                  </ForwardRef>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Available cars",
                  },
                  "ref": null,
                  "rendered": "Available cars",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-iwsKbI",
                      "isStatic": true,
                      "rules": Array [
                        "
  font-weight: 700;
  font-size: 18px;
  margin-top: 8px;
",
                      ],
                    },
                    "displayName": "styled.p",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-iwsKbI",
                    "target": "p",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Showing ",
                      1,
                      " of ",
                      undefined,
                      " ",
                      "result",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Showing ",
                    1,
                    " of ",
                    undefined,
                    " ",
                    "result",
                    " ",
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-jTzLTM",
                      "isStatic": true,
                      "rules": Array [
                        "
  font-size: 18px;
",
                      ],
                    },
                    "displayName": "styled.p",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-jTzLTM",
                    "target": "p",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-VigVT",
                  "isStatic": true,
                  "rules": Array [
                    "
  width: 60%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-VigVT",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <FilterBox
                  handleOnChange={[MockFunction]}
                  options={
                    Array [
                      Object {
                        "label": "None",
                        "value": "",
                      },
                      Object {
                        "label": "Mileage - Ascending Order",
                        "value": "asc",
                      },
                      Object {
                        "label": "Mileage - Descending Order",
                        "value": "des",
                      },
                    ]
                  }
                  title="Sort by"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleOnChange": [MockFunction],
                  "options": Array [
                    Object {
                      "label": "None",
                      "value": "",
                    },
                    Object {
                      "label": "Mileage - Ascending Order",
                      "value": "asc",
                    },
                    Object {
                      "label": "Mileage - Descending Order",
                      "value": "des",
                    },
                  ],
                  "title": "Sort by",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gZMcBi",
                  "isStatic": true,
                  "rules": Array [
                    "
  margin-bottom: 24px;

  @media (min-width: 768px) {
    width: 40%;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gZMcBi",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gqjmRU",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gqjmRU",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "car": Object {
              "color": "red",
              "fuelType": "diesel",
              "mileage": Object {
                "number": 234,
              },
              "modelName": "1",
              "stockNumber": 2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": undefined,
            "onNext": [MockFunction],
            "onPrevious": [MockFunction],
            "previousPage": undefined,
            "totalPageCount": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    width: 66.666%;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            <ForwardRef>
              <ForwardRef>
                Available cars
              </ForwardRef>
              <ForwardRef>
                Showing 
                1
                 of 
                 
                result
                 
              </ForwardRef>
            </ForwardRef>
            <ForwardRef>
              <FilterBox
                handleOnChange={[MockFunction]}
                options={
                  Array [
                    Object {
                      "label": "None",
                      "value": "",
                    },
                    Object {
                      "label": "Mileage - Ascending Order",
                      "value": "asc",
                    },
                    Object {
                      "label": "Mileage - Descending Order",
                      "value": "des",
                    },
                  ]
                }
                title="Sort by"
              />
            </ForwardRef>
          </ForwardRef>
          <Card
            car={
              Object {
                "color": "red",
                "fuelType": "diesel",
                "mileage": Object {
                  "number": 234,
                },
                "modelName": "1",
                "stockNumber": 2,
              }
            }
          />
          <Pagination
            onNext={[MockFunction]}
            onPrevious={[MockFunction]}
          />
        </ForwardRef>,
        "color": "#CDCDCD",
        "delay": 0,
        "ready": true,
        "row": 7,
        "type": "media",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <ForwardRef>
                <ForwardRef>
                  Available cars
                </ForwardRef>
                <ForwardRef>
                  Showing 
                  1
                   of 
                   
                  result
                   
                </ForwardRef>
              </ForwardRef>
              <ForwardRef>
                <FilterBox
                  handleOnChange={[MockFunction]}
                  options={
                    Array [
                      Object {
                        "label": "None",
                        "value": "",
                      },
                      Object {
                        "label": "Mileage - Ascending Order",
                        "value": "asc",
                      },
                      Object {
                        "label": "Mileage - Descending Order",
                        "value": "des",
                      },
                    ]
                  }
                  title="Sort by"
                />
              </ForwardRef>
            </ForwardRef>,
            Array [
              <Card
                car={
                  Object {
                    "color": "red",
                    "fuelType": "diesel",
                    "mileage": Object {
                      "number": 234,
                    },
                    "modelName": "1",
                    "stockNumber": 2,
                  }
                }
              />,
            ],
            <Pagination
              onNext={[MockFunction]}
              onPrevious={[MockFunction]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef>
                  <ForwardRef>
                    Available cars
                  </ForwardRef>
                  <ForwardRef>
                    Showing 
                    1
                     of 
                     
                    result
                     
                  </ForwardRef>
                </ForwardRef>,
                <ForwardRef>
                  <FilterBox
                    handleOnChange={[MockFunction]}
                    options={
                      Array [
                        Object {
                          "label": "None",
                          "value": "",
                        },
                        Object {
                          "label": "Mileage - Ascending Order",
                          "value": "asc",
                        },
                        Object {
                          "label": "Mileage - Descending Order",
                          "value": "des",
                        },
                      ]
                    }
                    title="Sort by"
                  />
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <ForwardRef>
                      Available cars
                    </ForwardRef>,
                    <ForwardRef>
                      Showing 
                      1
                       of 
                       
                      result
                       
                    </ForwardRef>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Available cars",
                    },
                    "ref": null,
                    "rendered": "Available cars",
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-iwsKbI",
                        "isStatic": true,
                        "rules": Array [
                          "
  font-weight: 700;
  font-size: 18px;
  margin-top: 8px;
",
                        ],
                      },
                      "displayName": "styled.p",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-iwsKbI",
                      "target": "p",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "Showing ",
                        1,
                        " of ",
                        undefined,
                        " ",
                        "result",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Showing ",
                      1,
                      " of ",
                      undefined,
                      " ",
                      "result",
                      " ",
                    ],
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-jTzLTM",
                        "isStatic": true,
                        "rules": Array [
                          "
  font-size: 18px;
",
                        ],
                      },
                      "displayName": "styled.p",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-jTzLTM",
                      "target": "p",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    },
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-VigVT",
                    "isStatic": true,
                    "rules": Array [
                      "
  width: 60%;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-VigVT",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <FilterBox
                    handleOnChange={[MockFunction]}
                    options={
                      Array [
                        Object {
                          "label": "None",
                          "value": "",
                        },
                        Object {
                          "label": "Mileage - Ascending Order",
                          "value": "asc",
                        },
                        Object {
                          "label": "Mileage - Descending Order",
                          "value": "des",
                        },
                      ]
                    }
                    title="Sort by"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleOnChange": [MockFunction],
                    "options": Array [
                      Object {
                        "label": "None",
                        "value": "",
                      },
                      Object {
                        "label": "Mileage - Ascending Order",
                        "value": "asc",
                      },
                      Object {
                        "label": "Mileage - Descending Order",
                        "value": "des",
                      },
                    ],
                    "title": "Sort by",
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-gZMcBi",
                    "isStatic": true,
                    "rules": Array [
                      "
  margin-bottom: 24px;

  @media (min-width: 768px) {
    width: 40%;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-gZMcBi",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-gqjmRU",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
  }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-gqjmRU",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "car": Object {
                "color": "red",
                "fuelType": "diesel",
                "mileage": Object {
                  "number": 234,
                },
                "modelName": "1",
                "stockNumber": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentPage": undefined,
              "onNext": [MockFunction],
              "onPrevious": [MockFunction],
              "previousPage": undefined,
              "totalPageCount": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    width: 66.666%;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
